examples/step-101/step-101.cc

```diff
@@ -0,0 +1,650 @@
+    // No non-matching data is needed for this part.
+    FEValues<dim> fe_values(fe_lagrange,
+                            cell_quadrature,
+                            update_values | update_gradients |
+                              update_JxW_values | update_quadrature_points);
+
+    FEFaceValues<dim> surface_fe_values(fe_lagrange,
+                                        face_quadrature,
+                                        update_values | update_gradients |
+                                          update_normal_vectors |
+                                          update_JxW_values |
+                                          update_quadrature_points);
+
+
+    for (const auto &cell :
+         dof_handler.active_cell_iterators() |
+           IteratorFilters::ActiveFEIndexEqualTo(ActiveFEIndex::lagrange))
+      {
+        local_stiffness = 0;
+        local_rhs       = 0;
+
+        const double cell_side_length = cell->minimum_vertex_distance();
+        fe_values.reinit(cell);
+
+        // Standard volume integration for the Laplace operator
+        for (const unsigned int q : fe_values.quadrature_point_indices())
+          {
+            const Point<dim> &point = fe_values.quadrature_point(q);
+            for (const unsigned int i : fe_values.dof_indices())
+              {
+                for (const unsigned int j : fe_values.dof_indices())
+                  {
+                    local_stiffness(i, j) += fe_values.shape_grad(i, q) *
+                                             fe_values.shape_grad(j, q) *
+                                             fe_values.JxW(q);
+                  }
+                local_rhs(i) += rhs_function.value(point) *
+                                fe_values.shape_value(i, q) * fe_values.JxW(q);
+              }
+          }
+
+        // Now we loop over faces. If a face borders a cell with FE_Nothing
+        // (i.e., a cell outside the domain), we apply shifted boundary
+        // conditions.
+        for (const auto &face : GeometryInfo<dim>::face_indices())
+          {
+            // Skip faces that border cells with active finite elements
+            if (cell->neighbor(face)->active_fe_index() ==
+                ActiveFEIndex::lagrange)
+              continue;
+
+            surface_fe_values.reinit(cell, face);
+
+            // Get the neighboring cell for level set evaluation
+            auto neighbour = cell->neighbor(face);
+            typename DoFHandler<dim>::cell_iterator neighbours_lvl_set_cell(
+              &neighbour->get_triangulation(),
+              neighbour->level(),
+              neighbour->index(),
+              &level_set_dof_handler);
+
+            neighbours_lvl_set_cell->get_dof_indices(level_set_dof_indices);
+            level_set.extract_subvector_to(level_set_dof_indices.begin(),
+                                           level_set_dof_indices.end(),
+                                           dof_values_level_set.begin());
```

Comment:
`dof_values_level_set` is not used

Reply:
Removed, not needed when using NonMatching::ClosestSurfacePoint